### ISSUE TITLE
Error running Perl Error: process execution timed out

### ISSUE THREAD

**Issue by DanFreed** [steelbrain/linter#1404]
I love the look and functionality of the new Linter, but I've started getting these errors on a somewhat regular basis (but not all of the time):

linter-registry.js [sm]:144 [Linter] Error running Perl Error: Process execution timed out
    at /Users/dfreed/.atom/packages/linter-perl-check-syntax/node_modules/sb-exec/lib/index.js:92:18(anonymous function) @ linter-registry.js [sm]:144
6events.js:160 Uncaught Error: write EPIPE

The code being listed is on an SMB share, so I'm guessing it has something to do with how fast/slow that network connection is.  

Is there a way to increase the timeout?

**Comment by steelbrain** [comment]
Hi @DanFreed 

Please contact the author of `linter-perl-check-syntax` asking for a timeout increase, tell the author that they can do so by supplying `timeout: 60000` in `atom-linter`'s `exec`/`execNode` method

**Comment by maschwenk** [comment]
@steelbrain this also cropped up in rubocop. https://github.com/AtomLinter/linter-rubocop/issues/202 . Did something change about the timeout logic such that I need to change the `timeout` you mentioned?

**Comment by steelbrain** [comment]
@maschwenk It did, it was a bugfix, see https://github.com/steelbrain/atom-linter/blob/master/CHANGELOG.md#901

tldr; atom-linter was advertising 10 seconds timeout when those timeouts weren't working, we fixed the timeout enforcement and now everything's breaking lol

**Comment by maschwenk** [comment]
Right, but even then, it feels like if any single-file linting process takes > 10 seconds, something went wrong. Upping the timeout per linter seems like the wrong solution...

**Comment by Arcanemagus** [comment]
@maschwenk That's the entire reason the timeout was added in the first place, `linter-phpcs` has a nasty habit of crashing and never executing. Systems were ending up with 40+ processes running all the time before that was put in place.

**Comment by DanFreed** [comment]
My original issue was a result of Listing some fairly large Perl scripts that they and their libraries live on a slowish network share.  10 seconds simply wasn't long enough for the worst of them, and Perl linting needs to load the library files to function correctly.  I believe the Perl linter is doing a perl -c on the current buffer which really just compiles the script and it's dependencies.

**Comment by Arcanemagus** [comment]
@DanFreed Exactly, there are some legitimate use cases for longer running processes which is why some linters have exemptions able to be enabled. https://github.com/steelbrain/exec/issues/81 should completely remove the need for the timeout though.
